tools/roxie/extract-roxie-timings.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+
+'''
+/*#############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems(R).
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################ */
+'''
+
+import csv
+import sys
+import re
+import argparse
+
+if __name__ == "__main__":
+    allStats = dict(time=1, elapsed=1)
+    allServices = dict()
+
+    minTimeStamp = ''
+    maxTimeStamp = ''
+
+    completePattern = re.compile("COMPLETE: ([^ ]*)")
+    elapsedPattern = re.compile("complete in ([0-9]*) ms")
+    idPattern = re.compile(r"(\{[^}]*\})")
+    yearMonthDayPattern = re.compile("^[0-9]{4}-[0-9]{2}-[0-9]{2}")
+    minuteSecondPattern = re.compile("^([0-9]*)m([0-9.]*)s$")
+
+
+    parser=argparse.ArgumentParser()
+    parser.add_argument("filename")
+    parser.add_argument("--all", "-a", help="Combine all services into a single result", action='store_true')
+    parser.add_argument("--nosummary", "-n", help="Avoid including a summary", action='store_true')
+    parser.add_argument("--summaryonly", "-s", help="Only generate a summary", action='store_true')
+    args = parser.parse_args()
+    combineServices = args.all
+    suppressDetails = args.summaryonly
+    reportSummary = not args.nosummary or args.summayonly
+
+    csv.field_size_limit(0x100000)
+    with open(args.filename) as csv_file:
```

Comment:
I had to add:
    with open(args.filename, encoding='latin1') as csv_file:
to get it to work with another log file, due to some chars causing an error:
  File "/usr/lib64/python3.6/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa8 in position 7872: invalid start byte
Not sure yet which char or why.

Reply:
I think there are some no-printable chars in the log file - no idea why

Reply:
I get a feeling one of the
SYS:
or
NIC:
system stats log lines might have a rogue char in it ??  I will look more later.

Reply:
Could leave this as default utf-8 encoding but then add a try/catch ?